orderer/common/cluster/commauth.go

```diff
@@ -0,0 +1,339 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster
+
+import (
+	"context"
+	"encoding/asn1"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/internal/pkg/identity"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+)
+
+// AuthCommMgr implements the Communicator
+// It manages the client side connections and streams established with
+// the Cluster GRPC server and new Cluster service
+type AuthCommMgr struct {
+	Logger  *flogging.FabricLogger
+	Metrics *Metrics
+
+	Lock           sync.RWMutex
+	shutdown       bool
+	shutdownSignal chan struct{}
+
+	Chan2Members MembersByChannel
+	Connections  *ConnectionsMgr
+
+	SendBufferSize int
+	NodeID         uint64
```

Comment:
unfortunately, we are not guaranteed we have the same identifier across all channels. we need to replace this with an identity and then do a lookup each time for each channel.

Reply:
Updated